packages/caliper-core/lib/worker/rate-control/maxRate.js

```diff
@@ -0,0 +1,203 @@
+/*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+const RateInterface = require('./rateInterface.js');
+const Sleep = require('../../common/utils/caliper-utils').sleep;
+const Logger = require('../../common/utils/caliper-utils').getLogger('maxRate.js');
+
+/**
+ * Rate controller for driving at a maximum TPS.
+ */
+class MaxRate extends RateInterface {
+    /**
+     * Creates a new instance of the MaxRate class.
+     * @constructor
+     * @param {object} opts Options for the rate controller.
+     */
+    constructor(opts) {
+        super(opts);
+
+        // Map for TPS observations
+        this.observedTPS = new Map();
+        // Map for TPS settings
+        this.tpsSettings = new Map();
+        // MPS for observed stats
+        this.statisticsMap = new Map();
```

Comment:
Since the maps only contain a few specific keys, it might be more readable and efficient to use fix objects instead (with sensible default values):
```js
this.observedTPS = {
  previous: 0,
  current: 0,
  max: 0
}
```

Reply:
I like!